lib/cms/repo.ex

```diff
@@ -135,6 +136,27 @@ defmodule CMS.Repo do
     end
   end
 
+  @spec photo() :: Photo.t() | nil
+  def photo do
+    cached_value = do_photo()
+
+    if cached_value == :empty || cached_value == :error, do: nil, else: cached_value
+  end
+
+  @decorate cacheable(
+              cache: @cache,
+              key: "cms.repo|photos",
+              on_error: :nothing,
+              opts: [ttl: @ttl]
+            )
+  def do_photo do
+    case @cms_api.view("/cms/photos", []) do
+      {:ok, [api_data | _]} -> Photo.from_api(api_data)
+      {:ok, _} -> :empty
```

Comment:
oh interesting, I hadn't quite realized the banner follows this pattern. 🤔 

`:empty` will be cached, is that what we want? Alternatively we could return `nil` (which won't be cached - this is governed by the default match function in the cacheable decorator function ([doc](https://hexdocs.pm/nebulex/Nebulex.Caching.Decorators.html#module-shared-options))).

Reply:
Pulling this back out of resolved 😁

Per [this comment](https://github.com/mbta/dotcom/pull/2695#discussion_r2294828860), I think we actually _do_ want `:empty` to be cached, so that we're not bombarding the CMS re-requesting the image if there isn't one.

I think not caching `nil`/`:error`/etc is supposed to avoid caching things like connectivity errors or timeouts, but `:empty` in this context is a successful response - it's the CMS saying "there's no image right now".

Reply:
Now reverted back to what it was in the initial PR, thanks for raising this Josh!